src/codecs/decoding/decoder.rs

```diff
@@ -15,9 +15,12 @@ use crate::{
 /// messages.
 #[derive(Clone)]
 pub struct Decoder {
-    framer: Framer,
-    deserializer: Deserializer,
-    log_namespace: LogNamespace,
+    /// The framer being used.
+    pub framer: Framer,
```

Comment:
These changes generally look good, but I'm missing what is special about the `aws_s3` source that required these to be public where they weren't before. What necessitated this change (same with `newline_delimited.rs` above)?

Reply:
For `Decoder`, the way SQS utilizes the decoder is a bit unique since it always allows a "multiline" config. The framing and decoding are done separately with the "multiline" code happening in the middle. So the code needed access to the individual parts.

For `newline_delimited.rs`, this is the first source that uses this as the default. This default was kept for backwards compatibility since it was previously hard-coded.

I could have added methods to access / modify these values instead of making them public, but since these types are simple data containers, it doesn't matter too much.